### PR TITLE
feat(filter) add filters in traps listing page

### PR DESCRIPTION
**Pull request by loiclau**
# Pull Request Template

## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/configuration/configObject/traps/listTraps.php [hunk 1]
```diff
@@ -41,37 +41,71 @@
...
+if ($searchVendor) {
+    $rq .= " AND manufacturer_id = " . (int)$searchVendor;
 }
+if ($searchStatus) {
+    $rq .= " AND traps_status = '" . ($enumStatus) . "'";
+}
+
+$rq .= ' ORDER BY manufacturer_id, traps_name LIMIT ' . $num * $limit . ', ' . $limit;
 
 $dbResult = $pearDB->query($rq);
```
Comment (sc979) [inline www/include/configuration/configObject/traps/listTraps.php]: could you bound these variables, as they seems injectable

File: www/include/configuration/configObject/traps/listTraps.php [hunk 2]
```diff
@@ -106,6 +139,30 @@
 );
 $form->addElement('submit', 'Search', _("Search"), $attrBtnSuccess);
 
+$attrTrapsStatus = null;
+if ($searchStatus && $searchStatus != '') {
```
Comment (sc979) [inline www/include/configuration/configObject/traps/listTraps.php]:
```suggestion
if (!empty($searchStatus)) {
```

File: www/include/configuration/configObject/traps/listTraps.php [hunk 3]
```diff
@@ -126,8 +183,8 @@
         "RowMenu_name" => $trap["traps_name"],
         "RowMenu_link" => "?p=$p&o=c&traps_id={$trap['traps_id']}",
         "RowMenu_desc" => substr($trap["traps_oid"], 0, 40),
-        "RowMenu_status" => isset($tabStatus[$trap["traps_status"]])
-            ? $tabStatus[$trap["traps_status"]]
+        "RowMenu_status" => isset($tabStatus[($trap["traps_status"])])
```
Comment (sc979) [inline www/include/configuration/configObject/traps/listTraps.php]:
```suggestion
        "RowMenu_status" => $tabStatus[($trap["traps_status"])] ?? $tabStatus[3]
```

File: www/include/configuration/configObject/traps/listTraps.php [hunk 4]
```diff
@@ -126,8 +183,8 @@
...
+            ? $tabStatus[($trap["traps_status"])]
```
Comment (sc979) [inline www/include/configuration/configObject/traps/listTraps.php]:
```suggestion
```

File: www/include/configuration/configObject/traps/listTraps.php [hunk 5]
```diff
@@ -126,8 +183,8 @@
...
             : $tabStatus[3],
```
Comment (sc979) [inline www/include/configuration/configObject/traps/listTraps.php]:
```suggestion
```

File: www/include/configuration/configObject/traps/listTraps.php [hunk 6]
```diff
@@ -151,11 +208,11 @@
 );
 
 ?>
-<script type="text/javascript">
-    function setO(_i) {
-        document.forms['form'].elements['o'].value = _i;
-    }
-</script>
+    <script type="text/javascript">
```
Comment (sc979) [inline www/include/configuration/configObject/traps/listTraps.php]: no need to indent this part

File: www/include/configuration/configObject/traps/listTraps.php [hunk 7]
```diff
@@ -106,6 +139,30 @@
...
+
+$vendorResult = $pearDB->query("SELECT id, name FROM traps_vendor ORDER BY name, alias");
+$vendors = [];
+for ($i = 0; $vendor = $vendorResult->fetch(); $i++) {
+    $vendors[$vendor['id']] = $vendor['name'];
+}
+
+$attrTrapsVendor = null;
+if ($searchVendor) {
+    $vendorDefault = array($vendors[$searchVendor] => $searchVendor);
```
Comment (sc979) [inline www/include/configuration/configObject/traps/listTraps.php]:
```suggestion
    $vendorDefault = array(
        $vendors[$searchVendor] => $searchVendor
    );
```

File: www/include/configuration/configObject/traps/listTraps.php [hunk 8]
```diff
@@ -106,9 +155,33 @@
...
+    $attrTrapsVendor = array(
+        'defaultDataset' => $vendorDefault
+    );
+}
+$form->addElement('select2', 'vendor', "", $vendors, $attrTrapsVendor);
+
 // Fill a tab with a multidimensional Array we put in $tpl
 $elemArr = array();
-for ($i = 0; $trap = $dbResult->fetch(); $i++) {
+for ($i = 0; $trap = $stmt->fetchRow(); $i++) {
```
Comment (sc979) [inline www/include/configuration/configObject/traps/listTraps.php]:
```suggestion
for ($i = 0; $trap = $stmt->fetch(); $i++) {
```

File: www/include/configuration/configObject/traps/listTraps.php [hunk 9]
```diff
@@ -40,42 +40,92 @@
...
+    $queryValues[':manufacturer'] = (int)$searchVendor;
+}
+if ($searchStatus) {
+    $rq .= ' AND traps_status = :status ';
+    $queryValues[':status'] = $enumStatus;
 }
 
-$dbResult = $pearDB->query($rq);
-$rows = $pearDB->query("SELECT FOUND_ROWS()")->fetchColumn();
+$rq .= ' ORDER BY manufacturer_id, traps_name LIMIT ' . $num * $limit . ', ' . (int)$limit;
```
Comment (callapa) [inline www/include/configuration/configObject/traps/listTraps.php]:
```suggestion
$rq .= ' ORDER BY manufacturer_id, traps_name LIMIT ' . (int) ($num * $limit) . ', ' . (int)$limit;
```

**Pull request merged**